pkg/services/provisioning/provisioning.go

```diff
@@ -198,7 +207,7 @@ func (ps *ProvisioningServiceImpl) Run(ctx context.Context) error {
 		case <-ctx.Done():
 			// Root server context was cancelled so cancel polling and leave.
 			ps.cancelPolling()
-			return ctx.Err()
+			return nil
```

Comment:
Just curious why this was changed

Reply:
sorry, it took me a while to track down this old thread with a similar question: https://github.com/grafana/grafana/pull/65147#discussion_r1146010884

> good question! when the dskit service [RunningFn](https://github.com/grafana/dskit/blob/727cbc22c1febfb5a77d024e407ebd01b207f74d/services/basic_service.go#L20) returns ctx.Err() it makes it seem like the service was the source of the error. since the [default implementation](https://github.com/grafana/dskit/blob/727cbc22c1febfb5a77d024e407ebd01b207f74d/services/services.go#L12-L15) of RunningFn for NewIdleService is to not report the error from context, i have been following that example
>
> i'm not confident that is the correct approach at this point. thoughts?

Reply:
Ah yes this sounds familiar - thanks for that!